core/base/allocator.hpp

```diff
@@ -76,20 +76,116 @@ class ExecutorAllocator {
     ExecutorAllocator(std::shared_ptr<ExecType> exec) : exec_{std::move(exec)}
     {}
 
+    /**
+     * Copy-constructs an allocator.
+     *
+     * @param other  the other allocator
+     */
+    ExecutorAllocator(const ExecutorAllocator &other)
+        : exec_{other.get_executor()}
+    {}
+
+    /**
+     * Move-constructs an allocator.
+     *
+     * This is required to not modify the original allocator!
+     *
+     * @param other  the other allocator
+     */
+    ExecutorAllocator(ExecutorAllocator &&other) : exec_{other.get_executor()}
```

Comment:
This change was necessary as a consequence of the [allocator traits related to move-construction/assignment](https://en.cppreference.com/w/cpp/named_req/Allocator):
> `A a1(std::move(a))` Constructs a1 such that it equals the prior value of a. Does not throw exceptions. The value of a is unchanged and a1 == a.

Due to its element-proxy architecture, MSVC's debug standard library caused issues there since it moved out of the allocator which was used to allocate the vector's data, and fails when freeing with it again.

Reply:
Ah, thanks, I did not know that this was a requirement.